main.js

```diff
@@ -18,13 +18,16 @@ if (!c) {
 const paddleOne = new Paddle({ x: 10, y: 60, width: 20, height: 50, color: 'teal' });
 const paddleTwo = new Paddle({ x: canvas.width - 30, y: 60, width: 20, height: 50, color: 'red' });
 const ball = new Ball({ x: 100, y: 60, radius: 12, color: 'yellow' });
+const paddleOneControls = { up: false, down: false };
+const paddleTwoControls = { up: false, down: false };
+
 let ballSpeedVertical = 1.5;
 let ballSpeedHorizontal = 1.5;
 let paddleOneSpeed = 0;
 let paddleTwoSpeed = 0;
 let acceleration = 0.2;
 let friction = 0.01;
-let maxSpeed = 5;
+let border = 10;
```

Comment:
Personally instead of having a border, I would set the ball x or y to be either where it should be, or at the minimum.

Like if its at the top:

```js
ball.y = Math.max(ball.y, ball.radius)
```
Then use Math.min for when its at the bottm

Reply:
Took a different direction on this, you can recomment if you still don't like it.